src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h

```diff
@@ -70,12 +70,12 @@ namespace OpenMS
   {
 public:
 
-    inline static ModificationsDB * getInstance()
+    inline static ModificationsDB * getInstance(OpenMS::String unimod_file = "CHEMISTRY/unimod.xml", OpenMS::String psimod_file = "CHEMISTRY/PSI-MOD.obo")
     {
       static ModificationsDB * db_ = 0;
       if (db_ == 0)
       {
-        db_ = new ModificationsDB;
+        db_ = new ModificationsDB(unimod_file, psimod_file);
```

Comment:
I like @cbielow solution where we would keep a static bool in the object which stores whether the db was ever instantiated so that the application can throw an error if it already was instantiated. Basically as @cbielow suggested: add a `static bool is_instanciated_` as well as a function `bool isInstanciated() const` to the object